pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
You're still not getting what I am on about. I understand all that clearly. I purposely ignored the clear + set of the IFS since it is irrelevant for this. All that matters is the end result at the end of the ISR, not where in the ISR the IFS is set or cleared.

The case I am concerned about is when the INT is ACTIVE at the end of the ISR so that IFS ends up SET but the INT pin becomes INACTIVE shortly afterwards so that the IFS is SET when the INT is INACTIVE. The ISR is then executed because IFS is SET, but INT is INACTIVE not ACTIVE, so that call to the ISR is a SPURIOUS call that should be filtered out.

Your current arrangement relies on the INT pin ALWAYS being set to its INACTIVE state at a point BEFORE the test to see if IFS should be set or not, which might not always be the case.